lib/src/utils.dart

```diff
@@ -29,11 +29,17 @@ final Logger logger = new Logger('dependency_validator');
 
 String bulletItems(Iterable<String> items) => items.map((l) => '  * $l').join('\n');
 
-Iterable<File> listDartFilesIn(String dirPath) {
+Iterable<File> listDartFilesIn(String dirPath, List<String> excludedDirs) {
   if (!FileSystemEntity.isDirectorySync(dirPath)) return const [];
-  return new Directory(dirPath)
-      .listSync(recursive: true)
-      .where((entity) => entity is File && !entity.path.contains('/packages/') && entity.path.endsWith('.dart'));
+
+  return new Directory(dirPath).listSync(recursive: true).where((entity) {
+    if (entity is! File) return false;
+    if (entity.path.contains('/packages/')) return false;
```

Comment:
Path #nit; it would be more portable to do:
```dart
if (path.split(entity.path).contains('packages')) return false;
```

https://www.dartdocs.org/documentation/path/1.5.1/path/split.html